test/routs/dutiesTest.js

```diff
@@ -0,0 +1,177 @@
+const mocha = require("mocha");
+const http = require("http");
+const XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const expect = require('chai').expect;
+const configs = require('../../config');
+const fs = require("fs");
+const runServer = require("../../start").runServer;
+
+
+describe("duties api", function () {
+  const url = configs.serverUrl + 'duties';
+
+  before("duties api", function (done) {
+    runServer(function (server, clientOut, dbOut) {
+      this.mongoServer = server;
+      this.client = clientOut;
+      this.db = dbOut;
+      const soldiersCollection = db.collection('soldiers');
+      const dutiesCollection = db.collection('duties');
+      soldiersCollection.deleteMany({}, function (err, result) {
+        dutiesCollection.deleteMany({}, function (err, result) {
+          fs.readFile("test/routs/dutiesTestData.JSON", function (err, data) {
+            testData = JSON.parse(data);
+            done();
+          });
+        });
+      });
+    });
+  });
+
+  after("duties api", function (done) {
+    client.close();
```

Comment:
if either of these is async, then this is profoundly written wrong. if niether is, then you don't need after to be async in the first place.